packages/light-client/src/spec/validateLightClientUpdate.ts

```diff
@@ -0,0 +1,133 @@
+import {altair, Root, ssz} from "@lodestar/types";
+import bls from "@chainsafe/bls/switchable";
+import type {PublicKey, Signature} from "@chainsafe/bls/types";
+import {
+  FINALIZED_ROOT_INDEX,
+  FINALIZED_ROOT_DEPTH,
+  NEXT_SYNC_COMMITTEE_INDEX,
+  NEXT_SYNC_COMMITTEE_DEPTH,
+  MIN_SYNC_COMMITTEE_PARTICIPANTS,
+  DOMAIN_SYNC_COMMITTEE,
+  GENESIS_SLOT,
+} from "@lodestar/params";
+import {getParticipantPubkeys, sumBits} from "../utils/utils.js";
+import {isValidMerkleBranch} from "../utils/index.js";
+import {SyncCommitteeFast} from "../types.js";
+import {isFinalityUpdate, isSyncCommitteeUpdate, isZeroedHeader, isZeroedSyncCommittee, ZERO_HASH} from "./utils.js";
+import {ILightClientStore} from "./store.js";
+
+export function validateLightClientUpdate(
+  store: ILightClientStore,
+  update: altair.LightClientUpdate,
+  syncCommittee: SyncCommitteeFast
+): void {
+  // Verify sync committee has sufficient participants
+  if (sumBits(update.syncAggregate.syncCommitteeBits) < MIN_SYNC_COMMITTEE_PARTICIPANTS) {
+    throw Error("Sync committee has not sufficient participants");
+  }
+
+  // Sanity check that slots are in correct order
+  if (update.signatureSlot <= update.attestedHeader.slot) {
```

Comment:
The spec also include the `current_slot >= update.signature_slot` assertion. Is it a conscious decision to not include that?

Reply:
Yes, to remove the dependency on the clock from this function. That check is done on the caller of this function